plugins/score-card/src/components/ScoreCardTable/ScoreCardTable.tsx

```diff
@@ -44,56 +45,68 @@ const useScoringAllDataLoader = () => {
 
 type ScoreTableProps = {
   title?: string;
-  scores: SystemScoreExtended[];
+  scores: EntityScoreExtended[] | SystemScoreExtended[];
 };
 
 export const ScoreTable = ({ title, scores }: ScoreTableProps) => {
-  const columns: TableColumn<SystemScoreExtended>[] = [
+  const columns: TableColumn<EntityScoreExtended | SystemScoreExtended>[] = [
     {
       title: 'Name',
       field: 'systemEntityName',
-      render: systemScore =>
-        systemScore.catalogEntityName ? (
-          <Link
-            to={`/catalog/${systemScore.catalogEntityName.namespace}/${systemScore.catalogEntityName.kind}/${systemScore.catalogEntityName.name}/score`}
-            data-id={systemScore.systemEntityName}
-          >
-            {systemScore.systemEntityName}
-          </Link> // NOTE: we can't use EntityRefLink as it does not yet support navigating to "/score" (or other tab) yet
-        ) : (
-          <>{systemScore.systemEntityName}</>
-        ),
+      render: entityScore => {
+
+        if ((entityScore as SystemScoreExtended).catalogEntityName) {
+          return (<Link
+            to={`/catalog/${entityScore.catalogEntityName?.namespace}/${entityScore.catalogEntityName?.kind}/${entityScore.catalogEntityName?.name}/score`}
+            data-id={entityScore.systemEntityName}
+            >
+              {entityScore.systemEntityName}
+            </Link>) // NOTE: we can't use EntityRefLink as it does not yet support navigating to "/score" (or other tab) yet
+        }
+
+        if ((entityScore as EntityScoreExtended).entityRef) {
+          const ref = (entityScore as EntityScoreExtended).entityRef
```

Comment:
can't we build also `const ref` in case we have `SystemScoreExtended` entity? so we build the `<Link .../>` only on one place?